new-integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/IstioUtils.java

```diff
@@ -0,0 +1,262 @@
+    try {
+      result = exec(new String(getIngressPort), true);
+    } catch (Exception ex) {
+      logger.info("Exception in getIngressPort() {0}", ex);
+      return 0;
+    }
+    logger.info("getIngressPort: kubectl returned {0}", result.toString());
+    if (result.stdout() == null) {
+      return 0;
+    } else {
+      return new Integer(result.stdout());
+    }
+  }
+
+  /**
+   * Get the secure https ingress port of istio installation.
+   * @return secure ingress https port for istio-ingressgateway
+  */
+  public static int getIstioSecureIngressPort() {
+    ExecResult result = null;
+    StringBuffer getSecureIngressPort = null;
+    getSecureIngressPort = new StringBuffer("kubectl -n istio-system get service istio-ingressgateway ");
+    getSecureIngressPort.append("-o jsonpath='{.spec.ports[?(@.name==\"https\")].nodePort}'");
+    logger.info("getSecureIngressPort: kubectl command {0}", new String(getSecureIngressPort));
+    try {
+      result = exec(new String(getSecureIngressPort), true);
+    } catch (Exception ex) {
+      logger.info("Exception in getSecureIngressPort() {0}", ex);
+      return 0;
+    }
+    logger.info("getSecureIngressPort: kubectl returned {0}", result.toString());
+    if (result.stdout() == null) {
+      return 0;
+    } else {
+      return new Integer(result.stdout());
+    }
+  }
+
+  /**
+   * Get the tcp ingress port of istio installation.
+   * @return tcp ingress port for istio-ingressgateway
+   */
+  public static int getIstioTcpIngressPort() {
+    ExecResult result = null;
+    StringBuffer getTcpIngressPort = null;
+    getTcpIngressPort = new StringBuffer("kubectl -n istio-system get service istio-ingressgateway ");
+    getTcpIngressPort.append("-o jsonpath='{.spec.ports[?(@.name==\"tcp\")].nodePort}'");
+    logger.info("getTcpIngressPort: kubectl command {0}", new String(getTcpIngressPort));
+    try {
+      result = exec(new String(getTcpIngressPort), true);
+    } catch (Exception ex) {
+      logger.info("Exception in getTcpIngressPort() {0}", ex);
+      return 0;
+    }
+    logger.info("getTcpIngressPort: kubectl returned {0}", result.toString());
+    if (result.stdout() == null) {
+      return 0;
+    } else {
+      return new Integer(result.stdout());
+    }
+  }
+
+  /**
+   * Deploy the Http Istio Gateway and Istio virtual service.
+   * @param inputTemplateFile input template file 
+   * @param outputTemplateFile output template file 
+   * @param templateMap map containing template variable(s) to be replaced
+   * @return true if deployment is success otherwise false
+  */
+  public static boolean deployHttpIstioGatewayAndVirtualservice(
+       String inputTemplateFile, String outputTemplateFile, 
+       Map<String, String> templateMap) throws IOException {
+
+    logger.info("Copying  source file {0} to target file {1}",inputTemplateFile,outputTemplateFile);
+
+    Path srcFile = Paths.get(inputTemplateFile);
+    Path targetFile = Paths.get(outputTemplateFile);
```

Comment:
I think the calling method should only give the output file name and this method should create it in RESULTS_ROOT/testname. My preference is to have a separate method in utils to copy and update the yaml file (which could be a generic method if other tests want to use it) and that should return the Path where the new file was created which can then be an argument for the deploy method.